code/modules/spells/targeted/recall.dm

```diff
@@ -53,6 +53,8 @@
 /spell/targeted/bound_object/is_valid_target(var/obj/target)
 	if(!istype(target))
 		return 0
+	if(target.z == CENTCOMM_Z && get_dist(target,holder) >= range)
```

Comment:
Should probably just make it so that it's if the Z level is teleport blocked instead, not hardcoded to the centcomm z level.